### PR TITLE
Clean up kernels.cu

### PR DESCRIPTION
**Pull request by jeremykubica**
A bunch of small clean ups for kernels.cu:
1) Added a bunch of comments.
2) Removed commented out code.
3) Remove an unused parameter (centralMomLims)
4) Added a test for the GPU filtering code.

### CODE REVIEW

**Comment by jeremykubica** [comment]
Ran diff test to confirm no change in behavior.

**Review by PWhiddy — COMMENTED** [review]
It looks like when searchFilterImages kernel was created it was copied from searchImages. That means most of the code in each of those functions is duplicated, which makes updating the code more work (as you had to make many of the same changes twice in this PR).

So I'm curious if both methods are still actually used/needed? 

Also just a general FYI for future cuda dev and refactoring - we haven't done this yet, but it is possible to factor common kernel code into functions. The functions which are called from kernels (```__global__ ``` functions) just need the ```__device__``` qualifier on their definition. You may already know this, just thought I'd mention 🙂

**Comment by PWhiddy** [comment]
Other than those comments, overall looks good!

File: search/src/kernels.cu [hunk 1]
```diff
@@ -241,39 +246,42 @@ __global__ void searchImages(int trajectoryCount, int width, int height,
                 //psiSum += -0.1;
```
[the remark this answers is not in the document]
Comment (jeremykubica) [reply search/src/kernels.cu]: Removed.

File: search/src/kernels.cu [hunk 2]
```diff
@@ -241,39 +246,42 @@ __global__ void searchImages(int trajectoryCount, int width, int height,
...
 
             currentT.obsCount++;
-            psiSum += cPsiPhi.x;// < NO_DATA/2 /*== NO_DATA* / ? 0.0 : cPsiPhi.x;//min(cPsi,0.3);
-            phiSum += cPsiPhi.y;// < NO_DATA/2 /*== NO_DATA* / ? 0.0 : cPsiPhi.y;
-            //if (psiSum <= 0.0 && i>4) break;
+            psiSum += cPsiPhi.x;
+            phiSum += cPsiPhi.y;
         }
         // Just in case a phiSum is zero
         //phiSum += phiSum*1.0005+0.001;
```
[the remark this answers is not in the document]
Comment (jeremykubica) [reply search/src/kernels.cu]: Removed.

**Pull request merged**